### PR TITLE
Added traffic selector policies field to virtual network gateway connection properties

### PR DESCRIPTION
**Pull request by ejdelrio**
## Checklist

- [x] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Network/Network/help/New-AzIpsecTrafficSelectorPolicy.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+## SYNTAX
+
+```
+New-AzIpsecTrafficSelectorPolicy -LocalAddressRanges <String[]> -RemoteAddressRanges <String[]>
+ [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
+```
+
+## DESCRIPTION
+{{ Fill in the Description }}
```
Comment (number213) [inline src/Network/Network/help/New-AzIpsecTrafficSelectorPolicy.md]: Please fill in the description (the same comment for other parts of this file)

File: src/Network/Network/help/New-AzIpsecTrafficSelectorPolicy.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.Network.dll-Help.xml
+Module Name: Az.Network
+online version:
```
Comment (number213) [inline src/Network/Network/help/New-AzIpsecTrafficSelectorPolicy.md]:
```suggestion
online version: https://docs.microsoft.com/en-us/powershell/module/az.network/new-azipsectrafficselectorpolicy
```

File: src/Network/Network/ChangeLog.md [hunk 3]
```diff
@@ -51,6 +51,12 @@
 * Adjusted validation logic for Ipv6 IP Prefix to check for correct IPv6 prefix length.
 * Updated Get-AzVirtualNetworkSubnetConfig: Added parameter set to get by subnet resource id.
 * Updated description of Location parameter for AzNetworkServiceTag
+* Added support for the adding traffic selectors to Virtual Network Gateway Connections
```
Comment (number213) [inline src/Network/Network/ChangeLog.md]: This should be added on top

**Comment by ejdelrio** [comment]
> 1. Please also update help for `New-AzureRmVirtualNetworkGatewayConnection` and `Set-AzureRmVirtualNetworkGatewayConnection`
> 2. There are unrelated changes in documentation. Can you rebase your PR again `network-septmber` to leave them out?

@number213 

1. Updated the MD files for Virtual Network Gateway Connection
2. Re-based my commits to exclude documentation from I pulled from master

**Comment by ejdelrio** [comment]
@number213 it seems that @msJinLei and @isra-fel are both OOF for the next week. Is there a way to change the assignee for this?

**Comment by number213** [comment]
@ejdelrio done, Mark will be the reviewer during holidays in China

File: src/Network/Network/help/New-AzIpsecTrafficSelectorPolicy.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
```
Comment (number213) [inline src/Network/Network/help/New-AzIpsecTrafficSelectorPolicy.md]: You need to set URL or the CI check `Analyze` will keep failing

**Comment by number213** [comment]
@ejdelrio I've merge network-september into your branch to get the recent changes that should fix CI. But please pull and re-record your test (the new SDK uses API version 2019-08-01 while your recording is for 2019-07-01)

File: src/Network/Network/VirtualNetworkGatewayConnection/NewAzureRmTrafficSelectorPolicy.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            HelpMessage = "A collection of CIDR address ranges")]
+        [ValidateNotNullOrEmpty]
+        public virtual string[] LocalAddressRanges { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "A collection of CIDR address ranges")]
+        [ValidateNotNullOrEmpty]
+        public virtual string[] RemoteAddressRanges { get; set; }
```
Comment (markcowl) [inline src/Network/Network/VirtualNetworkGatewayConnection/NewAzureRmTrafficSelectorPolicy.cs]: Parameter names cannot be plural

File: src/Network/Network/VirtualNetworkGatewayConnection/NewAzureRmTrafficSelectorPolicy.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+{
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "IpsecTrafficSelectorPolicy"), OutputType(typeof(PSTrafficSelectorPolicy))]
+    public class NewAzureRmTrafficSelectorPolicyCommand : NetworkBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "A collection of CIDR address ranges")]
+        [ValidateNotNullOrEmpty]
+        public virtual string[] LocalAddressRanges { get; set; }
```
Comment (markcowl) [inline src/Network/Network/VirtualNetworkGatewayConnection/NewAzureRmTrafficSelectorPolicy.cs]: Parameter names cannot be plural

File: src/Network/Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs [hunk 7]
```diff
@@ -144,6 +144,12 @@ public class NewAzureVirtualNetworkGatewayConnectionCommand : VirtualNetworkGate
              HelpMessage = "A list of IPSec policies.")]
         public PSIpsecPolicy[] IpsecPolicies { get; set; }
 
+        [Parameter(
+             Mandatory = false,
+             ValueFromPipelineByPropertyName = true,
+             HelpMessage = "A list of traffic selector policies.")]
+        public PSTrafficSelectorPolicy[] TrafficSelectorPolicies { get; set; }
```
Comment (markcowl) [inline src/Network/Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs]: Parameter names cannot be plural

File: src/Network/Network/help/New-AzIpsecTrafficSelectorPolicy.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+## EXAMPLES
+
+### Example 1
+```powershell
+$trafficSelectorPolicy = New-AzIpsecTrafficSelectorPolicy -LocalAddressRanges ("10.10.10.0/24", "20.20.20.0/24") -RemoteAddressRanges ("30.30.30.0/24", "40.40.40.0/24")
+New-AzVirtualNetworkGatewayConnection -ResourceGroupName $rgname -name $vnetConnectionName -location $location -VirtualNetworkGateway1 $vnetGateway -LocalNetworkGateway2 $localnetGateway -ConnectionType IPsec -RoutingWeight 3 -SharedKey $sharedKey -UsePolicyBasedTrafficSelectors $true -TrafficSelectorPolicies ($trafficSelectorPolicy)
+```
+
+{{ Add example description here }}
```
Comment (markcowl) [inline src/Network/Network/help/New-AzIpsecTrafficSelectorPolicy.md]: You need to fill this out.

Comment (ejdelrio) [reply src/Network/Network/help/New-AzIpsecTrafficSelectorPolicy.md]: Added description

File: src/Network/Network/VirtualNetworkGatewayConnection/NewAzureRmTrafficSelectorPolicy.cs (hunk 6 above)
Comment (ejdelrio) [reply src/Network/Network/VirtualNetworkGatewayConnection/NewAzureRmTrafficSelectorPolicy.cs]: Changed to singlular

File: src/Network/Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs (hunk 7 above)
Comment (ejdelrio) [reply src/Network/Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs]: Changed to singular

File: src/Network/Network/VirtualNetworkGatewayConnection/NewAzureRmTrafficSelectorPolicy.cs (hunk 5 above)
Comment (ejdelrio) [reply src/Network/Network/VirtualNetworkGatewayConnection/NewAzureRmTrafficSelectorPolicy.cs]: Changed to singuar

File: src/Network/Network/Models/PSTrafficSelectorPolicy.cs [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+// ----------------------------------------------------------------------------------
+
+using Microsoft.WindowsAzure.Commands.Common.Attributes;
+
+namespace Microsoft.Azure.Commands.Network.Models
+{
+    public class PSTrafficSelectorPolicy
+    {
+        /// <summary>
+        /// CIDR formmated local address spaces
```
Comment (number213) [inline src/Network/Network/Models/PSTrafficSelectorPolicy.cs]:
```suggestion
        /// CIDR formatted local address spaces 
```
typo

File: src/Network/Network/Models/PSTrafficSelectorPolicy.cs [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    public class PSTrafficSelectorPolicy
+    {
+        /// <summary>
+        /// CIDR formmated local address spaces 
+        /// </summary>
+        [Ps1Xml(Target = ViewControl.Table)]
+        public string[] LocalAddressRanges { get; set; }
+
+        /// <summary>
+        /// CIDR formmated remote address spaces
```
Comment (number213) [inline src/Network/Network/Models/PSTrafficSelectorPolicy.cs]:
```suggestion
        /// CIDR formatted remote address spaces 
```
typo

**Pull request merged**